src/plugins/formulas/formulas.js

```diff
@@ -564,6 +564,24 @@ export class Formulas extends BasePlugin {
     this.hot.runHooks('afterFormulasValuesUpdate', changes);
   }
 
+  /**
+   * Validate changes.
+   *
+   * @private
+   * @param {Array} changes The values and location of applied changes.
+   */
+  validateChanges(changes) {
+    changes.forEach((change) => {
+      // Named expression won't have address.
+      if (change?.address?.sheet === this.sheetId) {
```

Comment:
Just one small thing, we're still doing this check twice (https://github.com/handsontable/handsontable/pull/8080/files#diff-75a3f65e47cad775c4d9cef1c98b6ef3ad0ad696c9aaa6b62eaf9fe45a09833cR557).